opal/mca/common/ucx/common_ucx.c

```diff
@@ -74,3 +74,76 @@ OPAL_DECLSPEC void opal_common_ucx_mca_pmix_fence(ucp_worker_h worker)
     }
 }
 
+
+static void opal_common_ucx_wait_all_requests(void **reqs, int *count_p,
+                                              ucp_worker_h worker)
+{
+    int i;
+
+    MCA_COMMON_UCX_VERBOSE(2, "waiting for %d disconnect requests", *count_p);
+    for (i = 0; i < *count_p; ++i) {
+        opal_common_ucx_wait_request(reqs[i], worker, "ucp_disconnect_nb");
+        reqs[i] = NULL;
+    }
+
+    *count_p = 0;
+}
+
+OPAL_DECLSPEC int opal_common_ucx_disconnect_eps(size_t count, size_t max_disconnect,
+                                                 ucp_worker_h worker,
+                                                 ucp_ep_h (*get_ep)(size_t num, size_t *vpid, void *arg),
```

Comment:
can't we just pass an array of endpoints to this function? a malloc on finalize is not so bad..
we have have the shift pattern (e.g `(i+me)%np`) in this function to have it both in pml and spml
we can also pass the array of ompi_proc_t* to print the vpid

Reply:
also rename this func to opal_common_ucx_del_procs